### DISCUSSION TITLE
Get user data from external OIDC through JWT CLAIMS

### DISCUSSION THREAD

**Discussion by reichert** [Budibase/budibase#8045] [Help]
Hi people, I already opened a similar thread some while ago (#6236) and would like some advice in my scenario:

**Environment:**
- Self hosted, docker-compose, version 2.0.6

**What I need?**
- For my users authenticated through OIDC (keycloak / jwt) I need to filter the rows of a datasources through a field (example user_external_id) in order to filter only his records of a table.
- I'd like to automatically give access to one or more apps according to a field in the JWT Payload (also know as JWT CLAIM)

**Can you help me evaluate the best approach of my first need? Here is my 3 options until now:**

1) manually after each subscription setup the user with a field in the Users table called user_external_id. This could then be used as {{ Current User.user_external_id }} in the datasource filter.
- Problems with this approach: not automated work

2) use the data as exposed in the thirdPartyProfile - which contains all the JWT CLAIMs (in other words: the custom JWT fields).
- Problems with this approach: I could find until now a way to access something like {{ Current User.thirdPartyProfile.user_external_id }}. I read something about that this is not accessible by design. Is there a reason for that?

3) Build an application to save through budibase API a new user, filling the field user_external_id through API.
- Problems with this approach: the API must be called before first login

**And for the second need?**
1) I have no idea in how authorize a user as BASIC of an arbitrary budibase app without using API. Any toughts?

Thanks!

**Comment by Rory-Powell** [comment]
Hey @reichert , thanks for raising this. For your first point on accessing the third party profile I have raised this issue https://github.com/Budibase/budibase/issues/8200

For your second point about automatic app access you are correct there is no such functionality right now. We plan to support this in future by providing support for SCIM: http://www.simplecloud.info/. We don't have an exact date for this, but it will be hopefully sometime in Q1/Q2 23. This will be a feature on the pro plan and above.
